### ISSUE TITLE
cli_test doesn't compile on Windows due to using 'sys/socket.h'

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-core#1292]
**Build Error Description**

Reported by @cwyyprog in https://github.com/bitshares/bitshares-core/issues/1290#issue-354321160:
```
C:\bitshares-core\tests\cli\main.cpp(41): fatal error C1083: Cannot open include file: 'sys/socket.h': No such file or
directory [C:\bitshares-core\tests\cli_test.vcxproj]
Done Building Project "C:\bitshares-core\tests\cli_test.vcxproj" (default targets) -- FAILED.
```

**Build Environment**
 - Host OS:             Windows
 - Host Physical RAM    
 - Source Branch/Tag:   master
 - OpenSSL Version:     
 - Boost Version:       
 - C++ Compiler:        MSVC 2015 update 1

## CORE TEAM TASK LIST
- [ ] Evaluate `Build Error`
- [ ] Provide build guidance
- [ ] <OR> Create `Bug Report`

**Comment by abitmore** [comment]
This is an issue of `master` branch, perhaps we need a hotfix.

**Comment by jmjatlanta** [comment]
I made a simple test program based on the SO article, and it works with VS 2017, and should work for 2015...
```
#include "stdafx.h"

#include <iostream>
#include <string>
#ifdef _WIN32
#ifndef _WIN32_WINNT
#define _WIN32_WINNT 0x0501
#endif
#include <winsock2.h>
#include <WS2tcpip.h>
#else
#include <sys/socket.h>
#include <arpa/inet.h>
#include <netdb.h>
#include <unistd.h>
#endif

int sockInit(void)
{
#ifdef _WIN32
	WSADATA wsa_data;
	return WSAStartup(MAKEWORD(1, 1), &wsa_data);
#else
	return 0;
#endif
}

int sockQuit(void)
{
#ifdef _WIN32
	return WSACleanup();
#else
	return 0;
#endif
}

int get_available_port()
{
	struct sockaddr_in sin;
	int socket_fd = socket(AF_INET, SOCK_STREAM, 0);
	if (socket_fd == -1)
		return -1;
	sin.sin_family = AF_INET;
	sin.sin_port = 0;
	sin.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
	if (::bind(socket_fd, (const struct sockaddr*)&sin, sizeof(struct sockaddr_in)) == -1)
		return -1;
	socklen_t len = sizeof(sin);
	if (getsockname(socket_fd, (struct sockaddr*)&sin, &len) == -1)
		return -1;
	return sin.sin_port;
}

int main()
{
	sockInit();
	int port = get_available_port();
	std::cout << "The port chosen is " << std::to_string(port) << "\n";
	sockQuit();
    return 0;
}
```

**Comment by jmjatlanta** [comment]
This *should* fix it. If someone will please test and let me know if there are any issues...

https://github.com/bitshares/bitshares-core/commit/a8c4d6a075b333fc114ce1a32969e355ae62b42e

**Comment by abitmore** [comment]
Fixed by #1305.
